htdocs/user/bank.php

```diff
@@ -340,7 +341,7 @@
 		$user->rights->salaries->read && (in_array($object->id, $childids) || $object->id == $user->id)
 		)
 	{
-		$salary = new PaymentSalary($db);
+		$salary = new Salary($db);
```

Comment:
PaymentSalary i think

Reply:
@eldy i think we should see salaries list here and not payment salaries list, then i'm going to update sql query like this :

-               $sql = "SELECT ps.rowid, ps.datesp, ps.dateep, ps.amount";
-               $sql .= " FROM ".MAIN_DB_PREFIX."payment_salary as ps";
-               $sql .= " WHERE ps.fk_user = ".$object->id;
-               $sql .= " AND ps.entity = ".$conf->entity;
-               $sql .= " ORDER BY ps.datesp DESC";

+               $sql = "SELECT s.rowid, s.datesp, s.dateep, s.amount";
+               $sql .= " FROM ".MAIN_DB_PREFIX."salary as s";
+               $sql .= " WHERE s.fk_user = ".$object->id;
+               $sql .= " AND s.entity = ".$conf->entity;
+               $sql .= " ORDER BY s.datesp DESC";

what do you think about that ?

Reply:
As a user, i would prefer to see the payment of salaries. The salary is a "virtual" object that represent an "Intent to pay a salary" and we already know the amount of this information into the field salary of the record card. As a salary people, i know there is an intent to pay my salary every month, but i am more interesting into the amount paid and when it was paid. But different users may have different point of view on this.
Also, in a future, the object salary will be used for example to make treasury analysis in future (by having all salaries record of the year prerecorded). So for future treasury analysis or accountancy planification, salary will be more interesting but for the "record of a user", the payment of salary of this users seems better for my point of view.

Reply:
@eldy ok no problem, i just did it.
I had to change "AllSalaries" link on employee tab of user card to go to payments list and then i also update salaries payments list title into "Salaries payments of lastname firstname" because this list doesn't already have filters